packages/apputils/src/toolbar.ts

```diff
@@ -195,6 +199,44 @@ class Toolbar<T extends Widget> extends Widget {
 export
 namespace Toolbar {
   /**
+   * Create a toolbar item for a command.
+   */
+  export
+  function toolbarButtonFromCommand(commands: CommandRegistry, id: string): ToolbarButton {
+    let oldClass = commands.className(id)
+    let button = new ToolbarButton({
+      onClick: () => { commands.execute(id); },
+      className: oldClass,
+      tooltip: Private.commandTooltip(commands, id),
+    });
+    // Ensure that we pick up relevant changes to the command:
+    function onChange(sender: CommandRegistry, args: CommandRegistry.ICommandChangedArgs) {
+      if (args.id !== id) {
+        return;  // Not our command
+      }
+      if (args.type === 'removed') {
+        // Dispose of button
+        button.dispose();
+      } else if (args.type === 'changed') {
+        // Update all fields (onClick is already indirected)
+        let newClass = sender.className(id);
+        if (newClass !== oldClass) {
+          button.removeClass(oldClass);
+          button.addClass(newClass);
+          oldClass = newClass;
+        }
+        button.node.title = Private.commandTooltip(commands, id);
+      }
+    }
+    commands.commandChanged.connect(onChange);
```

Comment:
It is only true in this case because the base widget class dispose method clears its signal data.